source/diffHandler.py

```diff
@@ -0,0 +1,146 @@
+			buf += _dmpProc.stdout.read(size - len(buf))
+		return [
+			line
+			for line in buf.decode("utf-8").splitlines()
+			if line and not line.isspace()
+		]
+	except Exception:
+		log.exception("Exception in DMP, falling back to difflib")
+		return _difflib(newText, oldText)
+
+
+def _difflib(newText: str, oldText: str) -> List[str]:
+	newLines = newText.splitlines()
+	oldLines = oldText.splitlines()
+	outLines = []
+
+	prevLine = None
+
+	for line in ndiff(oldLines, newLines):
+		if line[0] == "?":
+			# We're never interested in these.
+			continue
+		if line[0] != "+":
+			# We're only interested in new lines.
+			prevLine = line
+			continue
+		text = line[2:]
+		if not text or text.isspace():
+			prevLine = line
+			continue
+
+		if prevLine and prevLine[0] == "-" and len(prevLine) > 2:
+			# It's possible that only a few characters have changed in this line.
+			# If so, we want to speak just the changed section, rather than the entire line.
+			prevText = prevLine[2:]
+			textLen = len(text)
+			prevTextLen = len(prevText)
+			# Find the first character that differs between the two lines.
+			for pos in range(min(textLen, prevTextLen)):
+				if text[pos] != prevText[pos]:
+					start = pos
+					break
+			else:
+				# We haven't found a differing character so far and we've hit the end of one of the lines.
+				# This means that the differing text starts here.
+				start = pos + 1
+			# Find the end of the differing text.
+			if textLen != prevTextLen:
+				# The lines are different lengths, so assume the rest of the line changed.
+				end = textLen
+			else:
+				for pos in range(textLen - 1, start - 1, -1):
+					if text[pos] != prevText[pos]:
+						end = pos + 1
+						break
+
+			if end - start < 15:
+				# Less than 15 characters have changed, so only speak the changed chunk.
+				text = text[start:end]
+
+		if text and not text.isspace():
+			outLines.append(text)
+		prevLine = line
+
+	return outLines
+
+
+def _should_use_DMP(supports_dmp: bool = True):
+	import config
+	return supports_dmp and config.conf["terminals"]["useDMPWhenSupported"]
+
+
+def initialize():
+	global _dmpProc
+	if not _dmpProc:
+		log.debug("Starting diff-match-patch proxy")
+		if hasattr(sys, "frozen"):
+			dmp_path = (os.path.join(globalVars.appDir, "nvda_dmp.exe"),)
+		else:
+			dmp_path = (sys.executable, os.path.join(
+				globalVars.appDir, "..", "include", "nvda_dmp", "nvda_dmp.py"
+			))
+		_dmpProc = subprocess.Popen(
+			dmp_path,
+			creationflags=subprocess.CREATE_NO_WINDOW,
+			bufsize=0,
+			stdin=subprocess.PIPE,
+			stdout=subprocess.PIPE
+		)
+
+
+def terminate():
+	global _dmpProc
+	try:
+		if _dmpProc:
+			log.debug("Terminating diff-match-patch proxy")
+			_dmpProc.stdin.write(struct.pack("=II", 0, 0))  # Sentinal value
+	except Exception:
+		pass
+
+
+def diff(newText: str, oldText: str, supports_dmp: bool = True):
+	return _dmp(newText, oldText) if _should_use_DMP(supports_dmp=supports_dmp) else _difflib(newText, oldText)

```

Comment:
I guess I could have two instances (DMP and difflib) and choose the implementation in `LiveText.__init__`?

Reply:
Yes, that would be fine. At the moment it looks like `LiveText` is the base class that sets the default behavior (in your PR currently DMP), and then `WinConsole` -> `Terminal` -> `LiveText` overrides it to `difflib`. This seems risky, it would be safer for "known good" usages to opt-in. 

Rather than use a bool for construction config, it would be better to have an internal (overriddable) method on LiveText to get the diff impl. EG `_getDiffImpl`. This can then be overridden by `WinConsole`

Reply:
It is far, far more likely that `LiveText` objects *will* be compatible with DMP than not (non-enhanced legacy consoles are an edge case). I've added a comment.